erpnext/maintenance/doctype/maintenance_schedule/maintenance_schedule.py

```diff
@@ -229,26 +228,35 @@ def create_opportunity_from_schedule(for_date=None):
 	""", target_date, as_dict=1)
 
 	for schedule in schedule_data:
-		schedule_doc = frappe.get_doc('Maintenance Schedule', schedule.parent)
-		opportunity_doc = frappe.new_doc('Opportunity')
+		create_maintenance_opportunity(schedule.parent, schedule.name)
+
+
+@frappe.whitelist()
+def create_maintenance_opportunity(maintenance_schedule, row):
+	schedule_doc = frappe.get_doc('Maintenance Schedule', maintenance_schedule)
+	default_opportunity_type = frappe.get_cached_value("CRM Settings", None, "default_opportunity_type_for_schedule")
+	schedule = schedule_doc.getone('schedules', {'name': row})
```

Comment:
Add a validation if schedule is None... frappe.throw otherwise you'll get unhandled exception accessing property `scheduled_date` from None error